src/EnergyPlus/EarthTube.cc

```diff
@@ -405,23 +405,41 @@ namespace EarthTube {
 			}
 		}
 
-		// Check to make sure there is only on ventilation statement per zone
-		for ( Loop = 1; Loop <= TotEarthTube; ++Loop ) {
-			for ( Loop1 = Loop + 1; Loop1 <= TotEarthTube - 1; ++Loop1 ) {
-				if ( EarthTubeSys( Loop ).ZonePtr == EarthTubeSys( Loop1 ).ZonePtr ) {
-					ShowSevereError( cAlphaArgs( 1 ) + " is assigned to more than one " + cCurrentModuleObject );
-					ShowContinueError( "Only one such assignment is allowed." );
-					ErrorsFound = true;
-				}
-			}
-		}
+		CheckEarthTubesInZones( cAlphaArgs( 1 ), cCurrentModuleObject, ErrorsFound );
 
 		if ( ErrorsFound ) {
 			ShowFatalError( cCurrentModuleObject + ": Errors getting input.  Program terminates." );
 		}
 
 	}
 
+	
+	void
+	CheckEarthTubesInZones
+		( std::string const ZoneName, // name of zone for error reporting
+		  std::string const FieldName, // name of earth tube in input
+		  bool & ErrorsFound // Found a problem
+		)
+	{
+
+		int Loop;
```

Comment:
Slowly coerce toward defining variables as they are used instead of at the top of the function.  I know it goes against your years of training, so no worries, just a note.